### PR TITLE
debouncing in javascript

### PR DESCRIPTION
**Pull request by afrazchelsea**
Hey mentors,

I wrote about the concept of Debouncing in Javascript. Please have a look.

### CODE REVIEW

File: js/debouncing.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+```
+let counter = 1; // To track number of calls
+
+const searchData = () => {
+  // Let's say this function calls some API and gets us some data
+  console.log("Looking for data..", counter++);
+};
+
+// Debounce function
```
Comment (itaditya) [inline js/debouncing.md:48]: After the code add a section giving overview of what we are doing. Then explain the code with a dry run example like first user enters M then callDebounce is called then this happens then that and so on. Showing this for first character, second character and last character will make it very clear what's happening.

**Review by itaditya — CHANGES_REQUESTED** [review]
Overall post is good. One opportunity of improving is there and I've mentioned it.

File: js/debouncing.md (hunk 1 above)
Comment (afrazchelsea) [reply js/debouncing.md:48]: Hey Aditya, I made a few changes. Please check if it looks good now.

**Review by itaditya — APPROVED** [review]

File: js/debouncing.md (hunk 1 above)
Comment (itaditya) [reply js/debouncing.md:48]: approved

**Pull request merged**